### PR TITLE
Improve time taken by Github actions

### LINKED ISSUE #365: Improve the time taken by Github Actions on pull requests
Coming from https://github.com/opensearch-project/opensearch-clients/issues/53, update the workflows to reduce the amount of time taken by Github actions for integration tests.

### PR DESCRIPTION
**Pull request by VachaShah**
### Description
Improve time taken by Github actions as stated in https://github.com/opensearch-project/opensearch-clients/issues/53 and some workflow updates.

### Issues Resolved
Closes #365 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: .github/workflows/release-drafter.yml [hunk 1]
```diff
@@ -25,9 +25,10 @@ jobs:
           issue-body: "Please approve or deny the release of opensearch-java. **VERSION**: ${{ steps.get_data.outputs.version }} **TAG**: ${{ github.ref_name }}  **COMMIT**: ${{ github.sha }}"
           exclude-workflow-initiator-as-approver: true
       - name: Set up JDK 11
-        uses: actions/setup-java@v1
+        uses: actions/setup-java@v3
         with:
           java-version: 11
+          distribution: adopt
```
Comment (reta) [inline .github/workflows/release-drafter.yml:31]:
```suggestion
          distribution: temurin
```

File: .github/workflows/test-integration-unreleased.yml [hunk 2]
```diff
@@ -18,36 +18,55 @@ jobs:
...
-          - { opensearch_ref: '2.0', java: 11 }
           - { opensearch_ref: 'main', java: 11 }
           - { opensearch_ref: 'main', java: 17 }
     steps:
       - name: Set up JDK ${{ matrix.entry.java }}
-        uses: actions/setup-java@v1
+        uses: actions/setup-java@v3
         with:
           java-version: ${{ matrix.entry.java }}
+          distribution: adopt
```
Comment (reta) [inline .github/workflows/test-integration-unreleased.yml:28]:
```suggestion
          distribution: temurin
```

File: .github/workflows/test-integration.yml [hunk 3]
```diff
@@ -21,12 +21,13 @@ jobs:
           - { opensearch_version: 2.5.0, java: 11 }
     steps:
       - name: Set up JDK ${{ matrix.java }}
-        uses: actions/setup-java@v1
+        uses: actions/setup-java@v3
         with:
           java-version: ${{ matrix.entry.java }}
+          distribution: adopt
```
Comment (reta) [inline .github/workflows/test-integration.yml:27]:
```suggestion
          distribution: temurin
```

File: .github/workflows/test-unit.yml [hunk 4]
```diff
@@ -11,12 +11,13 @@ jobs:
         os: [ubuntu-latest, windows-latest, macOS-latest]
     steps:
       - name: Set up JDK ${{ matrix.java }}
-        uses: actions/setup-java@v1
+        uses: actions/setup-java@v3
         with:
           java-version: ${{ matrix.java }}
+          distribution: adopt
```
Comment (reta) [inline .github/workflows/test-unit.yml:17]:
```suggestion
          distribution: temurin
```

File: .github/workflows/release-drafter.yml (hunk 1 above)
Comment (reta) [reply .github/workflows/release-drafter.yml:31]: You could also add `cache: gradle` to use Gradle artifacts caching

```
        with:
          java-version: 11
          distribution: 'temurin'
          cache: gradle
```

**Review by reta — APPROVED** [review]

**Review by dblock — APPROVED** [review]

**Pull request merged**